ui/components/ui/jazzicon/jazzicon.component.js

```diff
@@ -9,56 +9,60 @@ const iconFactory = iconFactoryGenerator(jazzicon);
  * Wrapper around the jazzicon library to return a React component, as the library returns an
  * HTMLDivElement which needs to be appended.
  */
-export default class Jazzicon extends PureComponent {
-  static propTypes = {
-    address: PropTypes.string.isRequired,
-    className: PropTypes.string,
-    diameter: PropTypes.number,
-    style: PropTypes.object,
-    tokenList: PropTypes.object,
-  };
 
-  static defaultProps = {
-    diameter: 46,
-  };
+function Jazzicon({
+  address,
+  className,
+  diameter = 46,
+  style,
+  tokenList = {},
+}) {
+  const container = useRef();
 
-  container = createRef();
+  useEffect(() => {
+    const addIcon = () => {
+      const imageNode = iconFactory.iconForAddress(
+        address,
+        diameter,
+        tokenList[address.toLowerCase()],
+      );
 
-  componentDidMount() {
-    this.appendJazzicon();
-  }
+      container.current?.appendChild(imageNode);
+    };
 
-  componentDidUpdate(prevProps) {
-    const { address: prevAddress, diameter: prevDiameter } = prevProps;
-    const { address, diameter } = this.props;
+    const removeIcon = () => {
+      container.current && (container.current.innerHTML = '');
+    };
 
-    if (address !== prevAddress || diameter !== prevDiameter) {
-      this.removeExistingChildren();
-      this.appendJazzicon();
-    }
-  }
+    removeIcon();
```

Comment:
If you return this function, instead of calling it, it'll be called only when the useEffect is re-run, before the new effect is ran. You may then remove the addIcon method and just run that code as the effect itself.